libs/@blockprotocol/graph/src/internal/mutate-subgraph.ts

```diff
@@ -50,103 +54,128 @@ export const addKnowledgeGraphEdgeToSubgraphByMutation = (
 };
 
 /**
- * Looking to build a subgraph? You probably want `import { buildSubgraph } from @blockprotocol/graph/stdlib`
+ * Looking to build a subgraph? You probably want {@link buildSubgraph} from `@blockprotocol/graph/stdlib`
  *
- * This MUTATES the given `Subgraph` by adding the given outwardEdge to the entity at the specified atTime.
+ * This MUTATES the given {@link Subgraph} by adding a given list of entities to the vertices, creating any link edges
+ * that are **directly implied** by any link entities in the list (see note below).
  * Mutating a Subgraph is unsafe in most situations – you should know why you need to do it.
  *
- * *Note*: This only adds edges as implied by the given entities, if the `Subgraph` is invalid at the time of method
- * call (e.g. by missing link endpoints), this will not loop through the vertex set to finish incomplete edges.
+ * *Note*: This only adds edges as implied by the given entities, if the {@link Subgraph} is invalid at the time of
+ * method call (e.g. by missing link endpoints), this will not loop through the vertex set to finish incomplete edges.
  *
- * @param subgraph – the subgraph to mutate by adding the provided entities
- * @param entities – the entity to add to the provided subgraph
+ * @param {Subgraph} subgraph – the subgraph to mutate by adding the provided entities
+ * @param {Entity[]} entities – the entity to add to the provided subgraph
  */
-export const addEntitiesToSubgraphByMutation = (
-  subgraph: Subgraph,
-  entities: Entity[],
+export const addEntitiesToSubgraphByMutation = <Temporal extends boolean>(
+  subgraph: Subgraph<Temporal>,
+  entities: Entity<Temporal>[],
 ) => {
+  /*
+   * @todo This assumes that the left and right entity ID of a link entity is static for its entire lifetime, that is
+   *   not necessarily going to continue being the case
+   */
   /* eslint-disable no-param-reassign -- We want to mutate the input here */
   const linkMap: Record<
     EntityId,
-    { leftEntityId: EntityId; rightEntityId: EntityId; earliestTime: Timestamp }
+    {
+      leftEntityId: EntityId;
+      rightEntityId: EntityId;
+      validIntervals: EntityValidInterval["validInterval"][];
+    }
   > = {};
 
   for (const entity of entities) {
-    const recordId = entity.metadata.recordId;
+    const entityId = entity.metadata.recordId.entityId;
+
+    const entityRevisionValidInterval: EntityValidInterval["validInterval"] =
+      subgraph.temporalAxes !== undefined
+        ? entity.metadata.temporalVersioning[
+            subgraph.temporalAxes.resolved.pinned.axis
+          ]
+        : {
+            start: { kind: "inclusive", limit: new Date(0).toISOString() },
```

Comment:
I was a bit unsure of what to put here but I'm not sure we do want the current date. This does feel like something we'd need to capture in a high-level definition of what a non-versioned subgraph actually looks like 🙃 as right now it's sort of implicitly designed in a few places across the code.

Assuming the thing is truly **non-versioned**, then it kind of makes sense to say the entities are just always valid, because then if you do end up passing in a timestamp or something in some logic somewhere else deep inside subgraph traversal, it will always return the entity.

I'm not suggesting that is what we want to do, that is perhaps hiding inconsistencies in implementation elsewhere and in a non-versioned subgraph you just should _not_ be able to pass in a timestamp (that _seems_ more ideal). But it does _seem_ like it's less likely to cause us hard to discover bugs in the meantime.